src/librustc_trait_selection/traits/error_reporting/suggestions.rs

```diff
@@ -1912,12 +1913,15 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
 
                 let parent_predicate = parent_trait_ref.without_const().to_predicate(tcx);
                 if !self.is_recursive_obligation(obligated_types, &data.parent_code) {
-                    self.note_obligation_cause_code(
-                        err,
-                        &parent_predicate,
-                        &data.parent_code,
-                        obligated_types,
-                    );
+                    // #74711: avoid segfaults
```

Comment:
I think this should mention "stack overflow", not "segfault". The latter is just a symptom on some platforms

Reply:
Reasonable, fixed.